src/composables/useLoop/index.ts

```diff
@@ -0,0 +1,67 @@
+import type { EventDispatcher, Raycaster, WebGLRenderer } from 'three'
+import { unref } from 'vue'
+import type { TresCamera, TresScene } from '../../types'
+import { useTresContext } from '../useTresContextProvider'
+import type { LoopCallback } from '../../core/loop'
+import type { SizesType } from '../useSizes'
+
+export interface LoopParams extends LoopCallback {
+  camera: TresCamera
+  scene: TresScene
+  renderer: WebGLRenderer
+  invalidate: () => void
+  advance: () => void
+  raycaster: Raycaster
+  controls: (EventDispatcher & { enabled: boolean })
+  sizes: SizesType
+}
+
+export type Fn = (params: LoopParams) => void
+
+export function useLoop() {
+  const {
+    camera,
+    scene,
+    renderer,
+    loop,
+    raycaster,
+    controls,
+    invalidate,
+    advance,
+  } = useTresContext()
+
+  function wrapCallback(cb: Fn) {
+    return (params: LoopCallback) => {
+      cb({ ...params, camera: unref(camera) as TresCamera, scene: unref(scene), renderer: unref(renderer), raycaster: unref(raycaster), controls: unref(controls), invalidate, advance })
```

Comment:
If possible, it's probably be best to avoid wrapping the callback. 

Currently, for every callback function in every frame, this code is recreating the argument object.

The argument object only needs to be created once.

Reply:
@andretchen0 how do we get the latest context if we are returning the plain values? if we called once on the onLoop are only going to get the snapshot of the context at that moment only